src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,60 @@
+package duke;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * The Parser class takes in a string, parses it to the required
+ * date and time format before returning it.
+ */
+public class Parser {
+
+    private String str;
+
+    /**
+     * Constructor for Parser class.
+     *
+     * @param str String that needs to be parsed.
+     */
+    public Parser(String str) {
+        this.str = str;
+    }
+
+    /**
+     * Returns a Task with a formatted date and time.
+     *
+     * @return parsedTask Task that has been formatted.
+     * @throws DukeException If task is empty.
+     * @throws ParseException If date or time of task does not follow the format.
+     */
+    public Task parse() throws DukeException, ParseException {
+        Task parsedTask = null;
+        String[] wordArray = str.split(" ");
+        String taskWithoutType = str.replace(wordArray[0], "").trim();
+
+        SimpleDateFormat format = new SimpleDateFormat("dd/MM/yyyy HHmm");
+        SimpleDateFormat formatter = new SimpleDateFormat("dd MMM yyyy, hh:mm a");
```

Comment:
This is slightly confusing. Why are there 2 different types of format here? 

`formatter` is quite vague. Maybe we can use `dateTimeFormat` instead? Hence, `dateTimeFormat.format(str)` is more obvious